File: 1134-armstrong-number/1134-armstrong-number.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isArmstrong = function(n) {
    let count = 0;
    let temp = n;
    while(temp>0){
        count++;
        temp = parseInt(temp/10);
    }
    let temp2 = n;
    let sum = 0;
    while(temp2>0){
        let val = temp2%10;
        sum += Math.pow(temp2%10,count);
        
        temp2 = parseInt(temp2/10);
    }
    
    return sum == n;

};

File: 1134-armstrong-number/1134-armstrong-number.java
class Solution {
    public boolean isArmstrong(int n) {
        int totalNum = (int) (Math.log10(n)+1);
        int temp = n;
        int sum = 0;
        while(temp>0){
            int val = temp%10;
            sum += (Math.pow(val%10,totalNum));
            temp = temp/10;
        }
        
        return sum == n;
    }
}